backend/src/users/controllers/users-controller.ts

```diff
@@ -1,41 +1,102 @@
+import {
+	Body,
+	Controller,
+	Delete,
+	FileTypeValidator,
+	ForbiddenException,
+	Get,
+	Param,
+	ParseFilePipe,
+	ParseIntPipe,
+	Patch,
+	UploadedFile,
+	UseFilters,
+	UseGuards,
+	UseInterceptors
+} from "@nestjs/common";
+import { UsersService } from "users/services/users.service";
+import { JwtAuthGuard } from "auth/guards/jwt-auth.guard";
+import { UpdateUserDto } from "users/model/dto/update-user.dto";
+import { UserReq } from "../user.decorator";
+import { UsersAvatarService } from "usersAvatar/services/usersAvatar.service";
+import { FileInterceptor } from "@nestjs/platform-express";
+import { diskStorage } from "multer";
+import { DeleteFileOnErrorFilter } from "filters/delete-file-on-error-filter";
 
 @Controller("users")
 class UsersController {
-    constructor(private readonly userService: UsersService) {
-    }
-
-    @UseGuards(JwtAuthGuard)
-    @Get()
-    async findAll() {
-        return this.userService.findAll();
-    }
-
-    @UseGuards(JwtAuthGuard)
-    @Get(":id")
-    async findOne(@UserReq("id") userId: number, @Param("id", ParseIntPipe) id: number) {
-        return this.userService.findOneById(id, userId);
-    }
-
-    @UseGuards(JwtAuthGuard)
-    @Delete(":id")
-    async delete(@UserReq("id") userId: number, @Param("id", ParseIntPipe) paramId: number) {
-        return this.userService.delete(userId, paramId);
-    }
-
-    @UseGuards(JwtAuthGuard)
-    @Patch(":id")
-    async update(
-        @Param("id", ParseIntPipe) id: number,
-        @UserReq("id") userId: number,
-        @Body() updateUserDto: UpdateUserDto
-    ) {
-        return this.userService.update(id, userId, updateUserDto);
-    }
+	constructor(
+		private readonly userService: UsersService,
+		private readonly usersAvatarService: UsersAvatarService
+	) {}
+
+	@UseGuards(JwtAuthGuard)
+	@Get()
+	async findAll() {
+		return this.userService.findAll();
+	}
+
+	@UseGuards(JwtAuthGuard)
+	@Get(":id")
+	async findOne(
+		@Param("id", ParseIntPipe) id: number,
+		@UserReq("id") userId: number
+	) {
+		if (id !== userId) throw new ForbiddenException();
+
+		return this.userService.findOneById(id);
+	}
+
+	@UseGuards(JwtAuthGuard)
+	@Delete(":id")
+	async delete(
+		@Param("id", ParseIntPipe) id: number,
+		@UserReq("id") userId: number
+	) {
+		if (userId !== id) throw new ForbiddenException();
+
+		return this.userService.delete(id);
+	}
+
+	@UseGuards(JwtAuthGuard)
+	@Patch(":id")
+	@UseFilters(new DeleteFileOnErrorFilter())
+	@UseInterceptors(
+		FileInterceptor("userAvatar", {
+			storage: diskStorage({
+				destination: "./avatars"
+			})
+		})
+	)
+	async update(
+		@Param("id", ParseIntPipe) id: number,
+		@UserReq("id") userId: number,
+		@Body() updateUserDto: UpdateUserDto,
+		@UploadedFile(
+			new ParseFilePipe({
+				validators: [new FileTypeValidator({ fileType: /(jpg|jpeg|png)$/ })],
+				fileIsRequired: false
+			})
+		)
+		file: Express.Multer.File
+	) {
+		if (id !== userId) throw new ForbiddenException();
+
```

Comment:
Is this flow corrent? No avatar passed = removing existing one? Just want to be sure.

Reply:
if user id from params didn't compare with this from JWT, i throw an Exception

Reply:
It's not what I wonder wbout. Look at all this method context. When no file is passed to controller You deleting already existed avatar. And I wonder if is it proper use case.

Reply:
That's alright, on frontend i will load avatar with whole user object, and that avatar will be sand to backend, if user remove it, null value will be send, in that situation, i know to remove it from DB and server storage.